test/interpret.jl

```diff
@@ -775,29 +783,14 @@ end
     @test (true === @interpret issue476())
 end
 
-const override_world = typemax(Csize_t) - 1
-macro unreachable(ex)
-    quote
-        world_counter = cglobal(:jl_world_counter, Csize_t)
-        regular_world = unsafe_load(world_counter)
-
-        $(Expr(:tryfinally, # don't introduce scope
-            quote
-                unsafe_store!(world_counter, $(override_world-1))
-                $(esc(ex))
-            end,
-            quote
-                unsafe_store!(world_counter, regular_world)
-            end
-        ))
-    end
+@noinline foobar() = (GC.safepoint(); 42)
+function run_foobar()
+    @eval foobar() = "nope"
+    @test_broken @interpret(foobar()) == foobar() == 42
+    nothing
```

Comment:
```suggestion
    return @interpret(foobar()), foobar()
```
and add `@test_broken run_foobar() == (42, 42) TODO #617` in the `"unreachable worlds"` testset.
Since it would be easier to debug things by running `run_foobar`.